Package.swift

```diff
@@ -129,19 +129,46 @@ func isDebuggingMain(_ isDebug: Bool) {
     }
 }
 
+func xcodeFolder() -> String {
+    (Executable("/usr/bin/xcode-select")("-p") ?? "").trimmed
+}
+
+func swiftCompilerVersion() -> String {
+    let versionLine = Executable(
+        "\(xcodeFolder().trimmed)/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend"
+    )("--version") ?? ""
+
+    let regex = try? NSRegularExpression(pattern: " \\d.\\d")
```

Comment:
Could use new regex stuff if tools version is new enough 🤔 but how can you detect that from within Package.swift?